syntax = "proto3";

import "google/protobuf/empty.proto";

package CameraAPI;
option go_package = "api.mooody.me/camapi";

service CameraService
{
    rpc SubscribeCameraStateChange(SubscribeCameraStateChangeRequest) returns (stream CameraState);
    rpc SetCameraState(SetCameraStateRequest) returns (google.protobuf.Empty);
}

message CameraState
{
    optional bool newState = 1;
    optional string ip4Address = 2;
    optional string ip6Address = 3;
    optional string motionEventId = 4;
    optional bytes imagePng = 5;
}

message Auth
{
    string secret = 1;
}

message SubscribeCameraStateChangeRequest
{
    Auth auth = 1;
}

message SetCameraStateRequest
{
    Auth auth = 1;
    CameraState state = 2;
}
